lib/grape/validations/validators/coerce.rb

```diff
@@ -68,9 +68,10 @@ def coerce_value(val)
         # define default values for structures, the dry-types lib which is used
         # for coercion doesn't accept nil as a value, so it would fail
         if val.nil?
-          return []      if type == Array || type.is_a?(Array)
+          return []      if type == Array
           return Set.new if type == Set
           return {}      if type == Hash
+          return val     if type.is_a?(Array)
```

Comment:
Should this be returning `val` (a.k.a. `nil`) in all cases here?

Reply:
I had that originally. I didn't see any specs fail. I was trying to minimize changes here and wasn't sure if there was a reason why the original code was checking `type.is_a?(Array)`.

Reply:
I added a test for coercion of `String`. I didn't see any issue with dry-types in the current form. Happy to change this to a `return val` if you think that's the right thing here.

Reply:
I think that's the right thing to do. We shouldn't have code that we don't understand ;) Let's find out the hard way if we broke something.

Reply:
Try to refactor out a `coerce_nil` method if you think it makes sense?

Reply:
Done!